File: greeter.py
import os
import urllib.parse

sent_query = os.environ['QUERY_STRING']
query_list = sent_query.split('=')
query_dict = urllib.parse.parse_qs(os.environ['QUERY_STRING'])

def greeter(name, surname):
    return('Hola ' + str(name).capitalize()+ ' ' + str(surname).capitalize() + '¿ como estas?')

input_name = str(query_dict['nombre'])[2:-2]
input_surname = str(query_dict['apellido'])[2:-2]

print("Tipo de contenido: text/html\n ")
print(str(greeter(input_name, input_surname)))

File: greeter.php
<?php

$html_template = '<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8">
        <meta name="viewport " content="width=device-width, initial-scale=1.0">
        <title>Greeter</title>
    </head>

    <body onload=""></body>
        <form action="" method="post">
            <input type="text" value="{VALUE_NAME}" placeholder="Name" name="name" id="name">&nbsp;&nbsp;&nbsp;
            <input type="text" value="{VALUE_SURNAME}" placeholder="Surname" name="surname" id="surname"><br><br>
            <input type="submit" value="Greet">
        </form>
    </body>
</html>';

echo $html_template;

$input_name = (array_key_exists('nombre', $_POST)) ? $_POST['nombre'] : "";
$input_surname = (array_key_exists('apellido',$_POST)) ? $_POST['apellido'] : "";

$html_output = str_replace("{VALUE_NAME}", $input_name, $html_template);
$html_output = str_replace("{VALUE_SURNAME}", $input_surname, $html_output);

echo $html_output;

$greeter_link = "http://localhost/py_api/greeter.py?name=" . $input_name . "&surname=" . $input_surname;
$greeter_data = file_get_contents($greeter_link);

echo"<br><br>" . $greeter_data;

?>